Open Judge System/Web/OJS.Web/Infrastructure/Filters/ValidateRemoteDataApiKeyFilter.cs

```diff
@@ -31,10 +32,13 @@ public class ValidateRemoteDataApiKeyFilter : IActionFilter<ValidateRemoteDataAp
             var apiKey = request.QueryString[ApiKeyQueryStringParamName] ?? request[ApiKeyQueryStringParamName];
 
             var isValidApiKey = !string.IsNullOrWhiteSpace(apiKey) &&
-                this.userManager.Users.Any(u => u.Id == apiKey && u.IsDeleted == false) &&
-                this.userManager.IsInRole(apiKey, GlobalConstants.AdministratorRoleName);
+                                this.userManager.Users.Any(u => u.Id == apiKey && u.IsDeleted == false) &&
+                                this.userManager.IsInRole(apiKey, GlobalConstants.AdministratorRoleName);
+
+            var isValidSecondaryApiKey = !string.IsNullOrWhiteSpace(apiKey) &&
+                                         apiKey == Settings.ApiKeySecondary;
 
-            if (!isValidApiKey)
+            if (!isValidApiKey && !isValidSecondaryApiKey)
```

Comment:
The check `string.IsNullOrWhiteSpace(apiKey)` is made twice. I suggest extracting two variables: `isApiKeyInSettings` and `isApiKeyInAdminUsers` and leave only `isValidApiKey`

```
var isValidApiKey =
  !string.IsNullOrWhiteSpace(apiKey) &&
  (isApiKeyInSettings || isApiKeyInAdminUsers)
```

That way if the key is in Settings, no unnecessary db call will be made.